### PR TITLE
12779 static embed maps

### PR DESCRIPTION
**Pull request by elenatorro**

### CODE REVIEW

File: lib/assets/javascripts/cartodb/common/public_navigation_view.js [hunk 1]
```diff
@@ -92,16 +95,29 @@ module.exports = cdb.core.View.extend({
...
-        var result = response.responseJSON;
-        this.viewModel.set({
-          likes: result.likes,
-          liked: result.liked
-        });
-      }.bind(this));
+      this.isLikeAnimated = true;
+
+      if (this.viewModel.get('liked')) {
+        this.client.delete([likeURL], function (error, response, data) {
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb/common/public_navigation_view.js]: let's extract the handler into a function in order to DRY.

File: lib/assets/javascripts/cartodb/common/public_navigation_view.js [hunk 2]
```diff
@@ -92,16 +95,29 @@ module.exports = cdb.core.View.extend({
...
+          if (error) {
+            return;
+          }
+          this.viewModel.set({
+            likes: data.likes,
+            liked: data.liked
+          });
+        }.bind(this));
+      } else {
+        this.client.post([likeURL], function (error, response, data) {
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb/common/public_navigation_view.js]: use it here as well.

Comment (elenatorro) [reply lib/assets/javascripts/cartodb/common/public_navigation_view.js]: 👍

File: lib/assets/javascripts/cartodb/embed_map/embed_map_details_view.js [hunk 3]
```diff
@@ -67,17 +69,29 @@ module.exports = cdb.core.View.extend({
...
-        }
+      this.isLikeAnimated = true;
 
-        var result = response.responseJSON;
-        this.viewModel.set({
-          likes: result.likes,
-          liked: result.liked
-        });
-      }.bind(this));
+      if (this.viewModel.get('liked')) {
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb/embed_map/embed_map_details_view.js]: Maybe we can extract it to its own module, to be called from different views.

Comment (elenatorro) [reply lib/assets/javascripts/cartodb/embed_map/embed_map_details_view.js]: Good idea!

**Pull request merged**